### PR TITLE
implement UsePreviousParameterValues per #170

### PR DESCRIPTION
**Pull request by tavisrudd**
Implements #170. I'll follow up with tests on this branch. @dannosaur can you give it a spin in the meantime?

### CODE REVIEW

**Comment by dannosaur** [comment]
@tavisrudd you're a legend! I'm actually in a position where I'm just about to implement CI for a project built using iidy, so I'll give this a thorough rundown next week on a test stack and let you know how things go.

**Comment by dannosaur** [comment]
It doesn't work. From what I can see, it's because with `for (const paramName in ..)`, `paramName` is actually an index rather than the parameter's name. It seems that this needs to be changed to use `array.forEach((paramName, index) => { .. });`.

Furthermore, `input.Parameters` isn't keyed by the parameter's name, but rather is just a simple array of objects. One must inspect each object individually, checking whether `ParameterKey == paramName` before doing the override.

I have managed to fix this in e303f3c, and it appears to work as expected.

**Comment by tavisrudd** [comment]
Teaches me to quickly fire stuff off late in the afternoon with not enough
thought and no testing :) I’ll try your fix later today. Thanks

**Comment by tavisrudd** [comment]
This looks good btw, I'm just holding off until I add some tests.

File: README.md [hunk 1]
```diff
@@ -202,6 +202,8 @@ data for creating or updating a CloudFormation stack.
 | StackPolicy | Location, `https`, or `s3` location of CloudFormation policy document | `policy.json` |
 | ResourceTypes | List of [allowed resource types to create in CloudFormation stack](https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_CreateStack.html) | `['AWS::EC2::*']` |
 | CommandsBefore | List of commands to run before `create-` and `update-stack` commands | `['make build']` |
+| UsePreviousTemplate | for updates, [boolean defaulting to false](https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_UpdateStack.html)
+| UsePreviousValueParameters | list of parameter names for updates, [equivalent to UsePreviousValue](https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_Parameter.html)
```
Comment (jpb) [inline README.md]: This reads a little better:

```suggestion
| UsePreviousParameterValues | list of parameter names for updates, [equivalent to UsePreviousValue](https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_Parameter.html)
```

Comment (tavisrudd) [reply README.md]: 👍 I'll follow up with that.

**Pull request merged**